### PR TITLE
docs: break apart the ambient getting start guide (WIP)

### PR DESCRIPTION
**Pull request by peterj**
## Description

[WIP] Breaking up the Ambient Getting Started section into multiple smaller sections.

## Reviewers

- [ ] Ambient
- [x] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure
- [ ] Localization/Translation

TODOs:

- [x] make the tests work
- [x] update the bookinfo screenshot (once new docker images are published)
- [ ] show L7 metrics (for some reason I don't see them in Kiali)

### CODE REVIEW

**Comment by peterj** [comment]
CC @craigbox; I started working on this - not ready for review yet, but wanted to get the draft PR in.

**Comment by peterj** [comment]
The first draft is ready to review -- cc @craigbox

**Comment by peterj** [comment]
*working on tests*

File: content/en/docs/ambient/getting-started/setup/index.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+weight: 1
+owner: istio/wg-networking-maintainers
+test: yes
+---
+
+In this section you'll download Istio CLI (`istioctl`) and install Istio in ambient mode on your Kubernetes cluster. You'll need a Kubernetes cluster to proceed. If you don't have a cluster, you can use [kind](/docs/setup/platform-setup/kind) or any other [supported Kubernetes platform](/docs/setup/platform-setup).
+
+## 1. Download Istio CLI
+
+You'll use [Homebrew](https://brew.sh/) to download the Istio CLI:
```
Comment (dhawton) [inline content/en/docs/ambient/getting-started/setup/index.md:13]:
```suggestion
You'll use [Homebrew](https://brew.sh/) to download the Istio CLI:
```

Personally don't like doing this, we should be encouraging users to download the binary from us with a version that matches what their cluster is running. We should link to this as a pre-req, or mirror the same instructions for consistency. https://istio.io/latest/docs/setup/getting-started/#download

Comment (peterj) [reply content/en/docs/ambient/getting-started/setup/index.md:13]: One of the things I tried to achieve with these instructions is to simplify them and get the reader ready to go as fast as possible. Executing a download script that downloads a compressed file, unpacks it into the current folder then asks you to set up your dotfiles/path to make it useful doesn't help with simplicity. The fact that we're not using either Brew or have a simpler installation might be a separate thing that could be improved. 

Note that brew automatically and on a schedule checks for outdated packages and updates them. (e.g. the 1.22 was out on May 13th and the brew formula was updated on the same day as well) 

If the consensus is that we should revert to the current instructions, I can do that, but I think we could do much better with simplifying the installation experience.

**Comment by craigbox** [comment]
https://deploy-preview-15168--preliminary-istio.netlify.app/latest/docs/ambient/getting-started/

File: content/en/docs/ambient/getting-started/_index.md
Comment (craigbox) [inline content/en/docs/ambient/getting-started/_index.md:1]: needs the magic for not displaying subpage summaries

File: content/en/docs/ambient/getting-started/setup/index.md (hunk 1 above)
Comment (craigbox) [reply content/en/docs/ambient/getting-started/setup/index.md:13]: I support the goal of making this easier, but I also need to point out that using we can't assume people are using a Mac.  For now, until we decide a larger change, let's use the standard "download CLI" instructions.

File: content/en/docs/ambient/getting-started/setup/index.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+✔ CNI installed
+✔ Ztunnel installed
+✔ Installation complete
+{{< /text >}}
+
+You can verify the installed components using the following command:
+
+{{< text syntax=bash snip_id=verify_install >}}
+$ istioctl verify-install
+
```
Comment (craigbox) [inline content/en/docs/ambient/getting-started/setup/index.md:60]: we can elide most of the items from this list?

File: content/en/docs/ambient/getting-started/setup/index.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
+---
+title: Install Ambient
```
Comment (craigbox) [inline content/en/docs/ambient/getting-started/setup/index.md:2]:
```suggestion
title: Install Istio with support for ambient mode
```

File: content/en/docs/ambient/getting-started/setup/index.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+description: Download and Install Istio in Ambient mode.
```
Comment (craigbox) [inline content/en/docs/ambient/getting-started/setup/index.md:3]:
```suggestion
description: Download and install Istio with support for ambient mode.
```

File: content/en/docs/ambient/getting-started/setup/index.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+weight: 1
+owner: istio/wg-networking-maintainers
+test: yes
+---
+
+In this section you'll download Istio CLI (`istioctl`) and install Istio in ambient mode on your Kubernetes cluster. You'll need a Kubernetes cluster to proceed. If you don't have a cluster, you can use [kind](/docs/setup/platform-setup/kind) or any other [supported Kubernetes platform](/docs/setup/platform-setup).
```
Comment (craigbox) [inline content/en/docs/ambient/getting-started/setup/index.md:9]: terminology note: "with support for ambient mode"

Comment (craigbox) [reply content/en/docs/ambient/getting-started/setup/index.md:9]: (ensure the kind docs are revised as part of this PR)

File: content/en/docs/ambient/getting-started/setup/index.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+$ cd istio-{{< istio_full_version >}}
+$ export PATH=$PWD/bin:$PATH
+{{< /text >}}
+
+{{< /quote >}}
+
+Verify the CLI is installed correctly by running the following command:
+
+{{< text syntax=bash snip_id=none >}}
+$ istioctl version
```
Comment (craigbox) [inline content/en/docs/ambient/getting-started/setup/index.md:33]: https://github.com/istio/istio/issues/51296

File: content/en/docs/ambient/getting-started/secure-and-visualize/index.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+$ istioctl dashboard kiali
+{{< /text >}}
+
+Click on the Traffic Graph and you should see the Bookinfo application:
+
+{{< image link="./kiali-ambient-bookinfo.png" caption="Kiali dashboard" >}}
+
+{{< tip >}}
+If you don't see any traffic in Kiali, try refreshing the Bookinfo application in your browser a few times. This will generate some traffic, which you should see in Kiali soon.
+{{</ tip >}}
```
Comment (craigbox) [inline content/en/docs/ambient/getting-started/secure-and-visualize/index.md:64]: I found I didn't have any namespaces selected by default, and then I found that I hadn't had enough traffic to generate anything.

Should we tell people to set up a pod to ping `productpage` and then go make a coffee? Or perhaps before we tell them to install Kiali?  Want to maximise the chance that people see some traffic from the get-go.

File: content/en/docs/ambient/getting-started/secure-and-visualize/index.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+If you click on one of the services on the the dashboard, you can see the inbound and outbound traffic metrics gathered by Istio:
```
Comment (craigbox) [inline content/en/docs/ambient/getting-started/secure-and-visualize/index.md:66]: perhaps be more prescriptive about which service they should click, as per the above

File: content/en/docs/ambient/getting-started/enforce-auth-policies/index.md [hunk 9]
```diff
@@ -0,0 +1,121 @@
+---
+title: Enforce authorization policies
+description: Enforce Layer 4 and Layer 7 authorization policies in an ambient mesh.
+weight: 4
+owner: istio/wg-networking-maintainers
+test: yes
+---
+
+After you have added your application to the ambient mesh, you can secure application access using Layer 4 authorization policies.
```
Comment (craigbox) [inline content/en/docs/ambient/getting-started/enforce-auth-policies/index.md:9]: We haven't used the term 'ztunnel' once throughout, I wonder if we should introduce the concept to explain why people now need to think about waypoiints

File: content/en/docs/ambient/getting-started/enforce-auth-policies/index.md [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+  - from:
+    - source:
+        principals:
+        - cluster.local/ns/default/sa/bookinfo-gateway-istio
+EOF
+{{< /text >}}
+
+If you open the Bookinfo application in your browser (`http://localhost:8080/productpage`), you will see the product page, just as before. However, if you try to access the `productpage` service from a different service account, you should see an error.
+
+Let's try accessing Bookinfo application from a `sleep` pod:
```
Comment (craigbox) [inline content/en/docs/ambient/getting-started/enforce-auth-policies/index.md:40]: standard note that we need this to be called "client", not "sleep" going forward

File: content/en/docs/ambient/getting-started/enforce-auth-policies/index.md [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+Since the `sleep` pod is using a different service account, it will not have access the `productpage` service:
+
+{{< text bash >}}
+$ kubectl exec deploy/sleep -- curl -s "http://productpage:9080/productpage"
+command terminated with exit code 56
+{{< /text >}}
+
+## Enforce Layer 7 authorization policy
+
+To enforce Layer 7 policies, you first need a {{< gloss "waypoint" >}}waypoint proxy{{< /gloss >}} for the namespace. This proxy will handle all Layer 7 traffic entering the namespace.
```
Comment (craigbox) [inline content/en/docs/ambient/getting-started/enforce-auth-policies/index.md:55]: TODO correct for the fact it doesn't apply for traffic coming from gateways - bug @linsun about hairpinning

File: content/en/docs/ambient/getting-started/enforce-auth-policies/index.md [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+    - source:
+        principals:
+        - cluster.local/ns/default/sa/sleep
+    to:
+    - operation:
+        methods: ["GET"]
+EOF
+{{< /text >}}
+
+Note the `targetRefs` field is used to specify the target service for the authorization policy of a waypoint proxy. The rules section is similar as before, but this time we added the `to` section to specify the operation that is allowed.
```
Comment (craigbox) [inline content/en/docs/ambient/getting-started/enforce-auth-policies/index.md:97]: better links can go here once we land the auth policy doc updates in istio/istio.io#15072

File: content/en/docs/ambient/getting-started/manage-traffic/index.md [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+
+To confirm that roughly 10% of the of the traffic from 100 requests goes to `reviews-v2`, you can run the following command:
+
+{{< text syntax=bash snip_id=test_traffic_split >}}
+$ kubectl exec deploy/sleep -- sh -c "for i in \$(seq 1 100); do curl -s http://productpage:9080/productpage | grep reviews-v.-; done"
+{{< /text >}}
+
+You'll notice the majority of requests go to `reviews-v1`. You can confirm the same if you open the Bookinfo application in your browser and refresh the page multiple times. Notice the requests from the `reviews-v1` don't have any stars, while the requests from `reviews-v2` have black stars.
+
+## Next steps
```
Comment (craigbox) [inline content/en/docs/ambient/getting-started/manage-traffic/index.md:48]: Can we also have a section showing the metrics have upgraded from TCP only to TCP and HTTP in Kiali?

File: content/en/docs/ambient/getting-started/manage-traffic/index.md [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+
+This section concludes the Getting Started guide for ambient mode. Continue exploring the [ambient mode user guides](/docs/ambient/usage/) to learn more about Istio's features and capabilities.
```
Comment (craigbox) [inline content/en/docs/ambient/getting-started/manage-traffic/index.md:50]: link the Cleanup section somehow

File: content/en/docs/ambient/getting-started/cleanup/index.md
Comment (craigbox) [inline content/en/docs/ambient/getting-started/cleanup/index.md:1]: not reviewed as it wasn't linked yet

**Comment by craigbox** [comment]
I've broken the testing pipeline. (#15200)

At one stage this PR contained a `setup/index.md' with a matching `snips.sh`. I moved that content back into the main index.md but the tests think that it used to be there, so it adds it to the list and fails when it's not.

In order to unblock this I'm going to try a commit with just the empty directory, and then one with a no-op snips.sh if needed.

(On the plus side, now is a good time for someone else to learn how istio.io testing works in great depth)

File: content/en/docs/ambient/getting-started/setup/index.md (hunk 1 above)
Comment (dhawton) [reply content/en/docs/ambient/getting-started/setup/index.md]: ```(e.g. the 1.22 was out on May 13th and the brew formula was updated on the same day as well)```

This illustrates my point. istioctl being upgraded ahead of the cluster can introduce some unexpected or bad behavior. Given the upgrade trends we see when Mitch did his surveys, many users are at least N-1, some further. istioctl should not be on a different minor version.

> I support the goal of making this easier, but I also need to point out that using we can't assume people are using a Mac. For now, until we decide a larger change, let's use the standard "download CLI" instructions.



> I support the goal of making this easier, but I also need to point out that using we can't assume people are using a Mac. For now, until we decide a larger change, let's use the standard "download CLI" instructions.

Homebrew is not just for Macs. I use it on my Linux systems. My point is consistency across docs, and that the version of istioctl should match the version of Istio installed in the clusters, something `brew` cannot determine on its own.

Comment (peterj) [reply content/en/docs/ambient/getting-started/setup/index.md]: Are we expecting upgrading users to go through the "Getting Started" page? Perhaps we need a section that's a quick start guide or something aimed at new users?

My assumption was that someone going through the "getting started" page probably doesn't have Istio installed at all. But I might be wrong.

File: content/en/docs/ambient/getting-started/setup/index.md (hunk 2 above)
Comment (peterj) [reply content/en/docs/ambient/getting-started/setup/index.md]: In my notes, I wrote "perhaps we need a --short option for the verify-install command" :) 

But yes, I'll remove this as the initial installation already outputs the checkmarks and verifies everything is installed.

File: content/en/docs/ambient/getting-started/secure-and-visualize/index.md [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+## Add Bookinfo to the mesh
+
+You can enable all pods in a given namespace to be part of an ambient mesh by simply labeling the namespace:
+
+{{< text bash >}}
+$ kubectl label namespace default istio.io/dataplane-mode=ambient
+namespace/default labeled
+{{< /text >}}
+
+Congratulations! You have successfully added all pods in the default namespace to the ambient mesh. 🎉
```
[the remark this answers is not in the document]
Comment (peterj) [reply content/en/docs/ambient/getting-started/secure-and-visualize/index.md:20]: I agree, I am just afraid involving tcpdump might complicate things. Perhaps showing the padlock icon in Kiali or something.

File: content/en/docs/ambient/getting-started/setup/index.md (hunk 1 above)
Comment (dhawton) [reply content/en/docs/ambient/getting-started/setup/index.md]: What about after getting started? IE, if brew upgrades istioctl, but Istio in their env is not upgraded. Manifests are baked into istioctl, and they're specific to that minor version. Setting someone up for a bad time, imo.

Comment (peterj) [reply content/en/docs/ambient/getting-started/setup/index.md]: Brew won't automatically update it, user has to run update/upgrade, but I see your point. 

I am not necessarily advocating for Brew per-se, but for an easier way to install the CLI. E.g. even if it was an existing downloadScript that downloads the binary only and puts it in ~/.istio folder or something and doesn't require me to set the path and it doesn't "pollute" my current folder with yaml manifests.

File: content/en/docs/ambient/getting-started/secure-and-visualize/index.md (hunk 7 above)
Comment (peterj) [reply content/en/docs/ambient/getting-started/secure-and-visualize/index.md:64]: I don't think they need to go and make a coffee as it doesn't take that long. I'll add a line that tells them to send 100 requests or something like that to generate some traffic.

Comment (peterj) [reply content/en/docs/ambient/getting-started/secure-and-visualize/index.md:64]: note that for me, the "default" namespace was selected by default. I'll add a note.

File: content/en/docs/ambient/getting-started/enforce-auth-policies/index.md (hunk 12 above)
Comment (peterj) [reply content/en/docs/ambient/getting-started/enforce-auth-policies/index.md:97]: added the link.

File: content/en/docs/ambient/getting-started/enforce-auth-policies/index.md (hunk 11 above)
Comment (peterj) [reply content/en/docs/ambient/getting-started/enforce-auth-policies/index.md:55]: Do we want to go into more details here or just link to the L7 doc page instead?

File: content/en/docs/ambient/getting-started/secure-and-visualize/index.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (peterj) [reply content/en/docs/ambient/getting-started/secure-and-visualize/index.md:20]: I have a separate work item to do this.

File: content/en/docs/ambient/getting-started/cleanup/index.md
Comment (peterj) [reply content/en/docs/ambient/getting-started/cleanup/index.md:1]: it's in now.

File: content/en/docs/ambient/getting-started/secure-and-visualize/index.md [hunk 16]
```diff
@@ -55,19 +40,27 @@ You can access the Kiali dashboard by running the following command:
...
 
 {{< tip >}}
-If you don't see any traffic in Kiali, try refreshing the Bookinfo application in your browser a few times. This will generate some traffic, which you should see in Kiali soon.
+If you don't see the traffic graph, try re-sending the traffic to the Bookinfo application and make sure you have selected the **default** namespace in the **Namespace** dropdown in Kiali.
+
+To see the mTLS status between the services, click the **Display** dropdown and click **Security**.
 {{</ tip >}}
 
-If you click on one of the services on the the dashboard, you can see the inbound and outbound traffic metrics gathered by Istio:
+If you click on th line connecting two services on the the dashboard, you can see the inbound and outbound traffic metrics gathered by Istio.
```
Comment (dhawton) [inline content/en/docs/ambient/getting-started/secure-and-visualize/index.md]:
```suggestion
If you click on the line connecting two services on the the dashboard, you can see the inbound and outbound traffic metrics gathered by Istio.
```

File: content/en/docs/ambient/getting-started/secure-and-visualize/index.md [hunk 17]
```diff
@@ -55,19 +40,27 @@ You can access the Kiali dashboard by running the following command:
...
 
 {{< image link="./kiali-tcp-traffic.png" caption="L4 traffic" >}}
 
-In addition to securing your traffic with mTLS and providing telemetry, Istio has created a strong identity for each service (a SPIFFE ID). This identity can be used for creating authorization policies.
+In addition to the TCP metrics, Istio has created a strong identity for each service - a SPIFFE ID. This identity can be used for creating authorization policies.
```
Comment (dhawton) [inline content/en/docs/ambient/getting-started/secure-and-visualize/index.md:63]:
```suggestion
In addition to the TCP metrics, Istio has created a strong identity for each service: a SPIFFE ID. This identity can be used for creating authorization policies.
```

**Review by dhawton — COMMENTED** [review]
Couple of minor nits.

File: content/en/docs/ambient/getting-started/setup/index.md (hunk 1 above)
Comment (peterj) [reply content/en/docs/ambient/getting-started/setup/index.md]: Resolving this - we're keeping the existing installation method in the guide. If there are changes that we want to make, they should be made in the installer first.

File: content/en/docs/ambient/getting-started/manage-traffic/index.md (hunk 13 above)
Comment (peterj) [reply content/en/docs/ambient/getting-started/manage-traffic/index.md:48]: Resolving for now -- opened https://github.com/istio/istio.io/issues/15212 update.

File: content/en/docs/ambient/getting-started/secure-and-visualize/index.md (hunk 16 above)
Comment (peterj) [reply content/en/docs/ambient/getting-started/secure-and-visualize/index.md]: I think this was updated already.

**Comment by craigbox** [comment]
/test doc.test.profile-ambient

That's new

**Comment by craigbox** [comment]
No tests in this profile yet @kfaseela ?

**Comment by kfaseela** [comment]
Test was migrated in https://github.com/istio/istio.io/pull/15210 yet to be merged

**Comment by craigbox** [comment]
OK - I don't know the dependency order here but this PR is now being checked against the ambient test and it's failing, with no apparent error - because there are no tests using it that succeed?

**Comment by kfaseela** [comment]
The folder for running ambient-profile is created in the other PR, so not available here and failing. But the test currently is non blocking. So u can merge this PR

**Comment by kfaseela** [comment]
The first PR is waiting for another Istio PR, so not yet merged

**Review by craigbox — APPROVED** [review]

**Comment by craigbox** [comment]
approved it and the clean-up PR. Most others i saw had merged already. Let
me know what I can do. We will need to follow with a PR removing the
otherwise empty snips folder/file which will probably require forcing
through

**Pull request merged**